pkg/api/validation/validation.go

```diff
@@ -112,7 +112,7 @@ func ValidateReplicationControllerName(name string, prefix bool) (bool, string)
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
 func ValidateServiceName(name string, prefix bool) (bool, string) {
-	return nameIsDNS952Label(name, prefix)
+	return nameIsDNS1123Label(name, prefix)
```

Comment:
Sorry, services are properly DNS952 for now.  That's on purpose (1123 allows leading digits which we don't handle)